### PR TITLE
VPN-5865: Naomi Testing Fixes

### PR DESCRIPTION
**Pull request by oskirby**
## Description
I am taking a stab at reviewing VPN-5865, and I have been finding a few things that don't work on my machine. This is a collection of fixes that I have needed to get the branch working in my particular environment.

## Reference
Upstream PR #9257
JIRA issue: [VPN-5865](https://mozilla-hub.atlassian.net/browse/VPN-5865)

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

[VPN-5865]: https://mozilla-hub.atlassian.net/browse/VPN-5865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/platforms/linux/daemon/wireguardutilslinux.cpp
Comment (brizental) [inline src/platforms/linux/daemon/wireguardutilslinux.cpp:679]: The `inet` familly already includes both IPv4 and IPv6. Why is this necessary?

Comment (oskirby) [reply src/platforms/linux/daemon/wireguardutilslinux.cpp:679]: In this case it does not, `AF_INET` only covers IPv4 when it comes to the routing table.

Comment (brizental) [reply src/platforms/linux/daemon/wireguardutilslinux.cpp:679]: TIL!

**Review by brizental — APPROVED** [review]

**Comment by brizental** [comment]
@oskirby Is this ready to be merged in?

**Comment by oskirby** [comment]
@brizental I think these changes are good to go, though I haven't given an honest test to multihop and split tunnelling yet.

**Comment by brizental** [comment]
> @brizental I think these changes are good to go, though I haven't given an honest test to multihop and split tunnelling yet.

I will wait for you to give me the green light on these two feature before merging then.

**Comment by oskirby** [comment]
This is getting closer, I have split tunnelling working now.

Next up I'll see if I can get things to work with reverse path filtering enabled.

**Pull request merged**
